src/tests/endpoints.test.js

```diff
@@ -691,6 +691,65 @@ describe("Endpoints Test", () => {
 				userDocument = await findUserByEmailService(newUser.email);
 			});
 		});
+
+		describe("GET /api/users/marketing-emails/unsubscribe", () => {
+			it("should update receiveMarkentingEmails to false if user is found", async () => {
+
+				// Generate a random user.
+				const newUser = await createUser();
+
+				// Signup a new user to update later.
+				const signupRes = await supertest(app)
+				.post("/api/auth/signup")
+				.set('Accept', 'application/json')
+				.send({
+					...newUser,
+					cpassword: newUser.password
+				})
+				
+				expect(signupRes.statusCode).toEqual(201);
+				expect(signupRes.body).toHaveProperty("message")
+				expect(signupRes.body.message).toEqual("User created successfully");
+
+				// Internal Service Call to fetch ID of the user.
+				let userDocument = await findUserByEmailService(newUser.email);
+				const userId = userDocument._id;
+
+				const verifyRes = await supertest(app)
+					.patch(`/api/admin/users/markverified/${userId}`)
+					.set("Accept", "application/json")
+					.set("Authorization", `Bearer ${adminUser.access_token}`)
+					.send();
+				expect(verifyRes.status).toEqual(200);
+				expect(verifyRes.body).toHaveProperty("message");
+				expect(verifyRes.body.message).toEqual("User verified successfully");
+
+				const loginRes = await supertest(app)
+					.post("/api/auth/login")
+					.set("Accept", "application/json")
+					.send({
+						email: newUser.email,
+						password: newUser.password,
+					});
+
+				expect(loginRes.statusCode).toEqual(200);
+				expect(loginRes.body).toHaveProperty("access_token");
+				expect(loginRes.body).toHaveProperty("refresh_token");
+
+				newUser.access_token = loginRes.body.access_token;
+				newUser.refresh_token = loginRes.body.refresh_token;
+				
+			const res = await supertest(app)
+				.get(`/api/users/marketing-emails/unsubscribe/${userId}`)
```

Comment:
Why do you need `userId` here, fetch it from the `access_token`.

Make the route as `/api/users/marketing-emails/unsubscribe` without any path parameter

Reply:
@tharun634 We will need user id as path param because will share a link to unsubscribe in email too.
we don't have `access_token` from email

Reply:
@tharun634 so I keep the id in the path?

Reply:
Yes @Arbaz23 , sorry for the confusion, Ayush has now updated the issue. Please refer to the updated route mentioned in the issue and integrate it into your changes :)

`GET /api/users/:id/marketing-emails/unsubscribe`

https://github.com/MultiEmail/backend/issues/159#issue-1423970229

![image](https://user-images.githubusercontent.com/53267275/199241714-35cf1fe8-3622-4c9a-8444-2b24e4ae6048.png)